CADETProcess/optimization/optimizationProblem.py

```diff
@@ -2759,7 +2760,10 @@ def cached_steps(self):
     def cache_directory(self):
         """pathlib.Path: Path for results cache database."""
         if self._cache_directory is None:
-            _cache_directory = settings.working_directory / f'diskcache_{self.name}'
+            if 'XDG_CACHE_HOME' in os.environ:
+                _cache_directory = Path(os.environ['XDG_CACHE_HOME']) / 'CADET-Process' / 'cache' / f'diskcache_{self.name}'
```

Comment:
I guess we could leave out the `cache` subdir here since it's already specified as the `XDG_CACHE_HOME`.